lib/Backend/BackendUtils.cpp

```diff
@@ -187,9 +187,29 @@ runtime::RuntimeBundle::create(const IRFunction &F,
   }
   auto constantMaxSize = constantAllocator.getMaxMemoryUsage();
 
-  // Compute the offsets for Placeholders.
+  // Placeholders should be allocated in a order of Input|InputOutput|Output.
+  std::vector<Placeholder const *> inputOutputPlaceholders;
+  std::vector<Placeholder const *> outputPlaceholders;
+  std::vector<Placeholder const *> contiguousPlaceholders;
   for (auto &v : F.findPlaceholders()) {
-    // Get the WeightVar for each Placeholder to calculate offsets.
+    if (isInput(v)) {
```

Comment:
I think that `isInput` and `isOutput` checks are not quite correct currently. They should consider only uses in the current function rather than in the whole module.